lib/deas/sinatra_app.rb

```diff
@@ -58,15 +60,28 @@ def self.new(server_config)
         # routes
         server_config.routes.each do |route|
           send(route.method, route.path) do
-            route.run(
-              server_data,
-              RequestData.new({
-                :route_path => route.path,
-                :request    => request,
-                :response   => response,
-                :params     => params
+            begin
+              route.run(
+                server_data,
+                RequestData.new({
+                  :route_path => route.path,
+                  :request    => request,
+                  :response   => response,
+                  :params     => params
+                })
+              )
+            rescue StandardError => err
```

Comment:
@kellyredding - You might expand this to a few additional errors like we did for sanford/qs (via dat-worker-pool): https://github.com/redding/dat-worker-pool/blob/master/lib/dat-worker-pool/worker.rb#L14.

The `Timeout::Error` is probably the worst one because net/http throws it and it can be buried in other gems as well.